### PR TITLE
applications: nrf5340_audio: Create broadcast_source app

### PR DESCRIPTION
**Pull request by alexsven**
- Create unique streamctrl for broadcast_source
- Move Kconfigs pertaining to broadcast to its own Kconfig
- Move zbus linking to streamctrl

### CODE REVIEW

File: applications/nrf5340_audio/src/audio/CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,11 @@
 
 target_sources(app PRIVATE
 	       ${CMAKE_CURRENT_SOURCE_DIR}/audio_system.c
-	       ${CMAKE_CURRENT_SOURCE_DIR}/streamctrl.c
 	       ${CMAKE_CURRENT_SOURCE_DIR}/audio_datapath.c
 	       ${CMAKE_CURRENT_SOURCE_DIR}/sw_codec_select.c
 )
+
+if (CONFIG_BT_BAP_BROADCAST_SOURCE)
```
Comment (koffes) [inline applications/nrf5340_audio/src/audio/CMakeLists.txt:13]: Discuss. We can use the GATEWAY and CONFIG_TRANSPORT_BIS defines. What about BAP vs CAP defines?

**Review by koffes — APPROVED** [review]

File: applications/nrf5340_audio/src/bluetooth/Kconfig [hunk 2]
```diff
@@ -13,6 +13,8 @@ rsource "bt_management/Kconfig"
 #----------------------------------------------------------------------------#
 menu "Bluetooth audio"
 
+rsource "bt_stream/broadcast/Kconfig"
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/bluetooth/Kconfig:17]: Is it possible to rsource this only if AUDIO_DEV = 2 && TRANSPORT_BIS?
Meaning moving this check to here

File: applications/nrf5340_audio/src/audio/streamctrl_broadcast_source.c [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+#include <string.h>
+#include <errno.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/debug/stack.h>
+#include <zephyr/zbus/zbus.h>
+
+#include "nrf5340_audio_common.h"
+#include "broadcast_internal.h"
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/audio/streamctrl_broadcast_source.c]: We should probably move "_internal" from these .h files at some point

**Review by erikrobstad — APPROVED** [review]

File: applications/nrf5340_audio/src/audio/CMakeLists.txt (hunk 1 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/audio/CMakeLists.txt:13]: I think using the CONFIG_BT_BAP defines are a good solution now.
And then transition away from using the CONFIG_AUDIO_DEV and CONFIG_TRANSPORT configs.

File: applications/nrf5340_audio/src/bluetooth/Kconfig (hunk 2 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/Kconfig:17]: I can put it in an if TRANSPORT_BIS, since it is valid for both device types

**Pull request merged**